lib/sms/string_conversion.c

```diff
@@ -0,0 +1,531 @@
+{
+    /* GSM 7 bit Default Alphabet table */
+    0x40, 0xA3, 0x24, 0xA5, 0xE8, 0xE9, 0xF9, 0xEC,  /*  0- 7: @£$...        */
+    0xF2, 0xC7, 0x0A, 0xD8, 0xF8, 0x0D, 0xC5, 0xE5,  /*  8-15: ...LF..CR..Åå */
+
+    0x20, 0x5F, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,  /* 16-23: ._...         */
+    0x20, 0x20, 0x20, 0x20, 0xC6, 0xE6, 0xDF, 0xC9,  /* 24-31: ..Escape.ÆæßÉ */
+
+    0x20, 0x21, 0x22, 0x23, 0x20, 0x25, 0x26, 0x27,  /* 32-39: Space !"#¤%&' */
+    0x28, 0x29, 0x2A, 0x2B, 0x2C, 0x2D, 0x2E, 0x2F,  /* 40-47: ()*+,-./      */
+
+    0x30, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36, 0x37,  /* 48-55: 01234567      */
+    0x38, 0x39, 0x3A, 0x3B, 0x3C, 0x3D, 0x3E, 0x3F,  /* 56-63: 89:;<=>?      */
+
+    0xA1, 0x41, 0x42, 0x43, 0x44, 0x45, 0x46, 0x47,  /* 64-71: ¡ABCDEFG      */
+    0x48, 0x49, 0x4A, 0x4B, 0x4C, 0x4D, 0x4E, 0x4F,  /* 72-79: HIJKLMNO      */
+
+    0x50, 0x51, 0x52, 0x53, 0x54, 0x55, 0x56, 0x57,  /* 80-87: PQRSTUVW      */
+    0x58, 0x59, 0x5A, 0xC4, 0xD6, 0xD1, 0xDC, 0xA7,  /* 88-95: XYZÄÖÑÜ§      */
+
+    0xBF, 0x61, 0x62, 0x63, 0x64, 0x65, 0x66, 0x67,  /*  96-103: ¿abcdefg    */
+    0x68, 0x69, 0x6A, 0x6B, 0x6C, 0x6D, 0x6E, 0x6F,  /* 104-111: hijklmno    */
+
+    0x70, 0x71, 0x72, 0x73, 0x74, 0x75, 0x76, 0x77,  /* 112-119: pqrstuvw    */
+    0x78, 0x79, 0x7A, 0xE4, 0xF6, 0xF1, 0xFC, 0xE0,  /* 120-127: xyzäöñüà    */
+
+    /* GSM 7 bit Default Alphabet extension table:
+     * These codes are used for interpreting extended character codes
+     * following the "escape" code 0x1B. */
+    0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,  /* 128-135/ext 0-7      */
+    0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,  /* 136-143/ext 8-15     */
+
+    0x20, 0x20, 0x20, 0x20, 0x5E, 0x20, 0x20, 0x20,  /* 144-151/ext 16-23  ^ */
+    0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,  /* 152-159/ext 24-31    */
+
+    0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,  /* 160-167/ext 32-39    */
+    0x7B, 0x7D, 0x20, 0x20, 0x20, 0x20, 0x20, 0x5C,  /* 168-175/ext 40-47 {}\*/
+
+    0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,  /* 176-183/ext 48-55    */
+    0x20, 0x20, 0x20, 0x20, 0x5B, 0x7E, 0x5D, 0x20,  /* 184-191/ext 56-63 [~]*/
+
+    0x7C, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,  /* 192-199/ext 64-71 |  */
+    0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,  /* 200-207/ext 72-79    */
+
+    0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,  /* 208-215/ext 80-87    */
+    0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,  /* 216-223/ext 88-95    */
+
+    0x20, 0x20, 0x20, 0x20, 0x20, 0xA4, 0x20, 0x20,  /* 224-231/ext 96-103 € */
+    0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,  /* 232-239/ext 104-111  */
+
+    0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,  /* 240-247/ext 112-119  */
+    0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,  /* 248-255/ext 120-127  */
+};
+
+/*********************************************************************************************
+ *
+ * Function: string_conversion_ascii_to_gsm7bit
```

Comment:
Move the function documentations to string_convertion.h (and convert to @brief, @details, @param[in, out] etc)